src/core/HazelcastError.ts

```diff
@@ -422,11 +424,13 @@ export class HazelcastSqlException extends HazelcastError {
         originatingMemberId: UUID,
         code: number,
         msg: string,
+        suggestion : string | null = null,
```

Comment:
I think you should move this parameter to last, or places like `SqlService#rethrow` or `SqlService#toHazelcastSqlException` would fail (and I am not sure how the tests are passing? Even the TypeScript compiler should alert that, right?)

Reply:
when you provide default value to a parameter it is optional as well. So it is like `suggestion? : string | null`. The only difference is that `suggestion?`'s default value is undefined.

Reply:
But, if you call `new HazelcastSqlException(uuid, errorcode, msg, err)` the `err` will be set to `suggestion` and that is wrong

Reply:
Ah you are right, in rethrow methods it should be: 

new HazelcastSqlException(uuid, errorcode, msg, undefined, err)

passing undefined to a argument leaves its default value.